### PR TITLE
Add `raven.async.shutdowntimeout` option.

### PR DESCRIPTION
**Pull request by bretthoerner**
#209 

Quick addition. I'm not in love with adding another constructor to `AsyncConnection` but it felt like the safest way to prevent breakage (and make this optional) for anyone that is already doing something custom there.

### CODE REVIEW

File: raven/src/main/java/com/getsentry/raven/DefaultRavenFactory.java [hunk 1]
```diff
@@ -55,6 +55,10 @@
      */
     public static final String QUEUE_SIZE_OPTION = "raven.async.queuesize";
     /**
+     * Option for the graceful shutdown timeout of the async executor, in milliseconds.
+     */
+    public static final String SHUTDOWN_TIMEOUT = "raven.async.shutdowntimeout";
```
Comment (tkaemming) [inline raven/src/main/java/com/getsentry/raven/DefaultRavenFactory.java]: This should probably be `SHUTDOWN_TIMEOUT_OPTION` for consistency with the names above.

**Comment by tkaemming** [comment]
Would it make sense to allow `-1` as a magic number to totally disable the timeout?

**Comment by bretthoerner** [comment]
Fixed name and added support for -1.

**Comment by tkaemming** [comment]
Assuming that the tests pass, this looks good to me. 👍 

> I'm not in love with adding another constructor to `AsyncConnection` but it felt like the safest way to prevent breakage (and make this optional) for anyone that is already doing something custom there.

Yeah, I can't think of another good way to do that except for implementing an options object, which seems like overkill at this point.

**Comment by bretthoerner** [comment]
https://github.com/getsentry/raven-java/commit/a5b4ab5942857ecf19521a9709eeefa781575757

**Pull request closed without merging**